validator/keymanager/remote-web3signer/keymanager.go

```diff
@@ -64,8 +68,8 @@ func NewKeymanager(_ context.Context, cfg *SetupConfig) (*Keymanager, error) {
 }
 
 // FetchValidatingPublicKeys fetches the validating public keys from the remote server or from the provided keys.
-func (km *Keymanager) FetchValidatingPublicKeys(ctx context.Context) ([][48]byte, error) {
-	if km.publicKeysURL != "" {
+func (km *Keymanager) FetchValidatingPublicKeys(ctx context.Context) ([][fieldparams.BLSPubkeyLength]byte, error) {
+	if km.publicKeysURL != "" && len(km.providedPublicKeys) == 0 {
```

Comment:
Be sure to add a test case for this new logic. Otherwise, it might be accidentally reverted and all tests might pass.

Also, please amend the godoc to mention that it would only fetch public keys if there are no provided public keys already attached to the key manager.